src/main/java/spoon/reflect/path/impl/CtPathImpl.java

```diff
@@ -33,9 +33,70 @@ public <T extends CtElement> List<T> evaluateOn(CtElement... startNode) {
 		for (CtPathElement element : elements) {
 			filtered = element.getElements(filtered);
 		}
+		if (filtered.isEmpty()) {
```

Comment:
I would rather add an early-return here or/and maybe extract the rest in a different method (something like `evalueOnShadowModel` or the like?)

Reply:
Suggestion accepted.